homeassistant/components/sensor/fritzbox_netmonitor.py

```diff
@@ -0,0 +1,146 @@
+    'external_ip': ['External IP address', '', 'mdi:server-network'],
+    'uptime': ['Uptime', 's', 'mdi:clock'],
+    'bytes_sent': ['Bytes sent', 'bytes', 'mdi:server-network'],
+    'bytes_received': ['Bytes received', 'bytes', 'mdi:server-network'],
+    'transmission_rate_up': ['Upstream', 'bytes/s', 'mdi:server-network'],
+    'transmission_rate_down': ['Downstream', 'bytes/s', 'mdi:server-network'],
+
+    'max_byte_rate_up': ['Maximum upstream-rate', 'bytes/s',
+                         'mdi:server-network'],
+    'max_byte_rate_down': ['Maximum downstream-rate', 'bytes/s',
+                           'mdi:server-network'],
+    'max_bit_rate_up': ['Maximum upstream-rate', 'bits/s',
+                        'mdi:server-network'],
+    'max_bit_rate_down': ['Maximum downstream-rate', 'bits/s',
+                          'mdi:server-network'],
+}
+
+
+def setup_platform(hass, config, add_devices, discovery_info=None):
+    """Set up the fritzbox monitor sensors."""
+    # pylint: disable=import-error
+    import fritzconnection as fc
+
+    host = config[CONF_HOST]
+
+    try:
+        fstatus = fc.FritzStatus(address=host)
+    except (ValueError, TypeError):
+        fstatus = None
+
+    if fstatus is None:
+        _LOGGER.error('Failed to establish connection to FRITZ!Box '
+                      'with IP: %s', host)
+        return 1
+    else:
+        _LOGGER.info('Successfully connected to FRITZ!Box')
+
+    devices = []
+    for resource in config[CONF_RESOURCES]:
+        if 'arg' not in resource:
+            resource['arg'] = ''
+        devices.append(FritzboxMonitorSensor(fstatus,
+                                             resource[CONF_TYPE],
+                                             resource['arg']))
+
+    add_devices(devices)
+
+    entity_ids = (dev.entity_id for dev in devices)
+    group.Group.create_group(hass, GROUP_NAME_ALL_DEVICES, entity_ids, False)
+
+
+class FritzboxMonitorSensor(Entity):
+    """Implementation of a fritzbox monitor sensor."""
+
+    def __init__(self, fstatus, sensor_type, argument=''):
+        """Initialize the sensor."""
+        self._name = '{} {}'.format(SENSOR_TYPES[sensor_type][0], argument)
+        self.argument = argument
+        self.type = sensor_type
+        self._state = None
+        self._unit_of_measurement = SENSOR_TYPES[sensor_type][1]
+        self._fstatus = fstatus
+        self.update()
+
+    @property
+    def name(self):
+        """Return the name of the sensor."""
+        return self._name.rstrip()
+
+    @property
+    def icon(self):
+        """Icon to use in the frontend, if any."""
+        return SENSOR_TYPES[self.type][2]
+
+    @property
+    def state(self):
+        """Return the state of the device."""
+        return self._state
+
+    @property
+    def unit_of_measurement(self):
+        """Return the unit of measurement of this entity, if any."""
+        return self._unit_of_measurement
+
+    def update(self):
```

Comment:
Here you should update your data object. You are communicating with your router, you want to throttle the data retrieval, and catch exceptions caused by connection issues.

It looks like that you use the `systemmonitor` sensor platform as a start. This platform is not communication with a third party.